### ISSUE TITLE
Retweet text truncated

### ISSUE THREAD

**Issue by edsu** [DocNow/twarc-csv#30]
Should the `text` column include the full text of the original retweet instead of the truncated version?

**Comment by igorbrigadir** [comment]
Yes, this can be done. We have the referenced tweets. I was thinking of keeping as close as possible to the json but I think this just creates more problems and confusion. I'll add that along with the other referenced tweets changes

**Comment by edsu** [comment]
Yes, I definitely understand wanting to keep it like the JSON. But I think people converting the JSON to CSV will exepct to see the full text of the retweet.

**Comment by numeroteca** [comment]
I was going to report exactly this, truncated text in the `text` field i nretweets makes it more difficult to work with them when using CSV. Earlier @congosto mentioned that to me.

**Comment by igorbrigadir** [comment]
I haven't forgotten about this - just didn't get around to it yet. Will do in the next few days though.

**Comment by igorbrigadir** [comment]
Another related issue is with hashtags as lists, but this would be a separate thing: https://twittercommunity.com/t/problems-getting-the-correct-data-structure-and-full-text/159459/2

**Comment by igorbrigadir** [comment]
After a few attempts, i've a new version, but i'm not sure anymore if simply replacing the retweet text is a good idea, as this breaks all the other metadata for the retweet, and replacing the retweet metadata with the retweeted one seems pointless, when including the referenced_tweets inline as before gives you the data you need.. So alternatively we can replace the retweets entirely. Either way, this means treating retweets differently. Quoted tweets and replies should be treated separately too, and definitely should not be replacing tweets. I don't think we should "merge" quoted tweets in the same way at all.

Maybe this is something that should be better documented because it's more of a property of twitter data, not of a CSV representation.

For example:

This is a retweet:

```
{
  "text": "RT @quocleix: Primer: Searching for Efficient Transformers for Language Modeling\n\nhttps://t.co/RJ7IfdBvKo\n\nWe use evolution to design a new…",
  "entities": {
    "mentions": [
      {
        "start": 3,
        "end": 12,
        "username": "quocleix",
        "id": "989251872107085824"
      }
    ],
    "urls": [
      {
        "start": 82,
        "end": 105,
        "url": "https://t.co/RJ7IfdBvKo",
        "expanded_url": "http://arxiv.org/abs/2109.08668",
        "display_url": "arxiv.org/abs/2109.08668"
      }
    ]
  },
  "author_id": "495430242",
  "id": "1440297242162651139",
  "reply_settings": "everyone",
  "created_at": "2021-09-21T12:49:56.000Z",
  "public_metrics": {
    "retweet_count": 121,
    "reply_count": 0,
    "like_count": 0,
    "quote_count": 0
  },
  "source": "Twitter Web App",
  "referenced_tweets": [
    {
      "type": "retweeted",
      "id": "1440024037896245248"
    }
  ],
  "conversation_id": "1440297242162651139",
  "possibly_sensitive": false,
  "lang": "en"
}
```

and the corresponding referenced_tweet (the original tweet):

```
{
  "text": "Primer: Searching for Efficient Transformers for Language Modeling\n\nhttps://t.co/RJ7IfdBvKo\n\nWe use evolution to design a new Transformer variant, called Primer. Primer has a better scaling law, and is 3X to 4X faster for training than Transformer for language modeling. https://t.co/daoHnuQEjS",
  "author_id": "989251872107085824",
  "id": "1440024037896245248",
  "reply_settings": "everyone",
  "attachments": {
    "media_keys": [
      "3_1440021169722167300"
    ]
  },
  "created_at": "2021-09-20T18:44:19.000Z",
  "public_metrics": {
    "retweet_count": 121,
    "reply_count": 2,
    "like_count": 566,
    "quote_count": 22
  },
  "source": "Twitter Web App",
  "conversation_id": "1440024037896245248",
  "possibly_sensitive": false,
  "entities": {
    "urls": [
      {
        "start": 68,
        "end": 91,
        "url": "https://t.co/RJ7IfdBvKo",
        "expanded_url": "http://arxiv.org/abs/2109.08668",
        "display_url": "arxiv.org/abs/2109.08668"
      },
      {
        "start": 271,
        "end": 294,
        "url": "https://t.co/daoHnuQEjS",
        "expanded_url": "https://twitter.com/quocleix/status/1440024037896245248/photo/1",
        "display_url": "pic.twitter.com/daoHnuQEjS"
      }
    ]
  },
  "lang": "en"
}
```

Note the differences between `public_metrics` and `entities` in the retweet and original tweet. The original tweet has 2 URLs, a link and an Image, the retweet has the link but is missing the image, and if we replace the retweet text

`"RT @quocleix: Primer: Searching for Efficient Transformers for Language Modeling\n\nhttps://t.co/RJ7IfdBvKo\n\nWe use evolution to design a new…"`

 with the original

`"Primer: Searching for Efficient Transformers for Language Modeling\n\nhttps://t.co/RJ7IfdBvKo\n\nWe use evolution to design a new Transformer variant, called Primer. Primer has a better scaling law, and is 3X to 4X faster for training than Transformer for language modeling. https://t.co/daoHnuQEjS"`

Then we are missing the metadata for the `https://t.co/daoHnuQEjS` link - and the `public_metrics` (retweets only show `retweet_count` number ) and the retweet entities no longer match.

In order for things to work, the entire tweet has to be copied over, and only the `author` and `id` and `created_at` changed.

So the choices are:

* Do nothing: Keep just the result output, with truncated retweet text, and no referenced tweets in the CSV (this is the worst choice i think, right now it's like the current one with `--no-inline-referenced-tweets` this is what the main branch is on now)

* Inject the original tweet just before the retweet (the previous behavior `--inline-referenced-tweets` preserves all the data but confuses people - maybe do this only for retweets?)

* Replace entire retweet: drop all the retweet metadata and replace it with the original tweet metadata, apart from the `author` and `id`.

I'm thinking of 3: replacing entire tweet if it's a retweet, with an extra command line switch `--merge-retweets` / `--no-merge-retweets`

I'll post some examples of commands and outputs later to make this more clear.
